### PR TITLE
add support for online groups

### PR DESCRIPTION
**Pull request by s-cheng**
This PR adds support for online groups, and displays them in a separate list on the main Community/Events page.

---
When you create a group, checking the "This is an online group" checkbox will hide the "Location" field. For local (i.e. not online) groups, the location is still required.

Screen Shot 2021-11-08 at 2 55 15 PM

---
On the group page, I hid the map for online groups. I have an upcoming project to allow groups to upload a group picture to display at the top of this page (for local groups, we'll probably move the map down and make it smaller).

Screen Shot 2021-11-08 at 4 33 33 PM

We can also add the group picture to the list item component, to make this page look a bit more colorful.

Screen Shot 2021-11-08 at 4 32 50 PM

---
Based on some feedback from Catherine Low, I also added the group name to a couple places where we list event details. This is helpful for online events called something like "Social / Hangout".

Screen Shot 2021-11-08 at 4 29 34 PM

Screen Shot 2021-11-08 at 4 22 44 PM

### CODE REVIEW

File: packages/lesswrong/server/callbacks/localgroupCallbacks.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+getCollectionHooks("Localgroups").createValidate.add(async (validationErrors: Array<any>, {document: group}: {document: DbLocalgroup}) => {
+  if (!group.isOnline && !group.location)
+    throw new Error("Location is required for local groups");
+  
+  return validationErrors;
+});
+
+getCollectionHooks("Localgroups").updateValidate.add(async (validationErrors: Array<any>, {oldDocument, newDocument}: {oldDocument: DbLocalgroup, newDocument: DbLocalgroup}) => {
+  if (!newDocument.isOnline && !newDocument.location)
+    throw new Error("Location is required for local groups");
```
Comment (s-cheng) [inline packages/lesswrong/server/callbacks/localgroupCallbacks.ts:13]: This is how I handled the location requirement. I'm not sure if this error message is clear though - if you have any suggestions for how to refer to a non-online group besides "local group" let me know.

File: packages/lesswrong/server/notificationCallbacks.tsx [hunk 2]
```diff
@@ -663,7 +663,7 @@ getCollectionHooks("Posts").updateAsync.add(async function PostsEditMeetupNotifi
   if (
     (
       (!newPost.draft && oldPost.draft) || 
-      (newPost.mongoLocation && !newPost.mongoLocation) || 
+      (newPost.mongoLocation !== oldPost.mongoLocation) ||
```
Comment (s-cheng) [inline packages/lesswrong/server/notificationCallbacks.tsx:666]: I was just guessing here but seemed like a bug?

File: packages/lesswrong/components/localGroups/CommunityHome.tsx [hunk 3]
```diff
@@ -163,6 +168,15 @@ const CommunityHome = ({classes}: {
                 </PostsList2>}
               </AnalyticsContext>
             </SingleColumnSection>
+            
+            <SingleColumnSection>
+              <SectionTitle title="Online Groups">
+                {canCreateEvents && <GroupFormLink />}
```
Comment (jcmustin) [inline packages/lesswrong/components/localGroups/CommunityHome.tsx:174]: Minor consideration, but you could auto-populate the "This is an online group" checkbox by passing down an `isOnline` prop through GroupFormLink into GroupFormDialog, which then sets `prefilledProps={{isOnline}}}` on WrappedSmartForm. Not a huge deal either way.

**Review by jcmustin — APPROVED** [review]
Looks good to me! Also it's been a bit since I last poked around the community and events page, and it's looking great!

File: packages/lesswrong/components/localGroups/CommunityHome.tsx (hunk 3 above)
Comment (s-cheng) [reply packages/lesswrong/components/localGroups/CommunityHome.tsx]: I like this idea! Added :)

**Pull request merged**